api/maven-api-core/src/main/java/org/apache/maven/api/plugin/Log.java

```diff
@@ -0,0 +1,166 @@
+package org.apache.maven.api.plugin;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.function.Supplier;
+
+/**
+ * This interface supplies the API for providing feedback to the user from the <code>Mojo</code>, using standard
+ * <code>Maven</code> channels.<br>
+ * There should be no big surprises here, although you may notice that the methods accept
+ * <code>java.lang.CharSequence</code> rather than <code>java.lang.String</code>. This is provided mainly as a
+ * convenience, to enable developers to pass things like <code>java.lang.StringBuffer</code> directly into the logger,
+ * rather than formatting first by calling <code>toString()</code>.
+ */
+public interface Log
```

Comment:
What do you think about stuff of colorized messages from:
 
https://maven.apache.org/shared/maven-shared-utils/apidocs/org/apache/maven/shared/utils/logging/MessageBuilder.html

Reply:
What about enhancing it after this one is merged (and before it rewrites all maven code in branches ;))

Reply:
A  [`MessageBuilderFactory`](https://github.com/apache/maven/blob/bf4995e7f4c8ae473b441539824bf3712f82a553/api/maven-api-core/src/main/java/org/apache/maven/api/services/MessageBuilderFactory.java) is available from the `Session` to support those use cases.